google/cloud/storage/internal/object_read_streambuf.cc

```diff
@@ -22,12 +22,22 @@ namespace cloud {
 namespace storage {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 namespace internal {
+namespace {
+
+std::streamoff InitialOffset(ReadObjectRangeRequest const& request) {
+  if (request.HasOption<ReadLast>()) {
+    return -request.GetOption<ReadLast>().value();
+  }
+  return request.StartingByte();
+}
+
+}  // namespace
 
 ObjectReadStreambuf::ObjectReadStreambuf(
     ReadObjectRangeRequest const& request,
-    std::unique_ptr<ObjectReadSource> source, std::streamoff pos_in_stream)
+    std::unique_ptr<ObjectReadSource> source)
     : source_(std::move(source)),
-      source_pos_(pos_in_stream),
+      source_pos_(static_cast<std::streamoff>(InitialOffset(request))),
```

Comment:
Why do we need to static cast to `std::streamoff` when it looks like `InitialOffset()` already returns a `std::streamoff`?

Reply:
D'oh! That is what happens when you refactor to a function and forget to clean up the call site.